<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1080</width>
    <height>670</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QMdiArea" name="mdiAreaOriginal">
    <property name="geometry">
     <rect>
      <x>225</x>
      <y>10</y>
      <width>830</width>
      <height>610</height>
     </rect>
    </property>
   </widget>
   <widget class="QTreeWidget" name="listFiles">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>200</width>
      <height>270</height>
     </rect>
    </property>
    <column>
     <property name="text">
      <string>打开图像</string>
     </property>
    </column>
   </widget>
   <widget class="QTreeWidget" name="listPFiles">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>320</y>
      <width>200</width>
      <height>300</height>
     </rect>
    </property>
    <column>
     <property name="text">
      <string>处理图像</string>
     </property>
    </column>
   </widget>
   <widget class="QPushButton" name="clickDown">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>285</y>
      <width>32</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="sources.qrc">
      <normaloff>:/icon/resources/down.ico</normaloff>:/icon/resources/down.ico</iconset>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="clickUp">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>285</y>
      <width>32</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="sources.qrc">
      <normaloff>:/icon/resources/up.ico</normaloff>:/icon/resources/up.ico</iconset>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1080</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>文件</string>
    </property>
    <addaction name="actionOpenFile"/>
    <addaction name="actionCloseFile"/>
    <addaction name="actionSaveFile"/>
   </widget>
   <widget class="QMenu" name="menuArithm">
    <property name="title">
     <string>算数运算</string>
    </property>
    <addaction name="actionNegOps"/>
    <addaction name="separator"/>
    <addaction name="actionToGrayScale"/>
    <addaction name="actionToBinary"/>
    <addaction name="separator"/>
    <addaction name="actionArithmOpses"/>
   </widget>
   <widget class="QMenu" name="menuGeometryOps">
    <property name="title">
     <string>几何变换</string>
    </property>
    <widget class="QMenu" name="menu">
     <property name="title">
      <string>镜像</string>
     </property>
     <addaction name="actionFlipHorizontal"/>
     <addaction name="actionFlipVertical"/>
    </widget>
    <addaction name="actionMove"/>
    <addaction name="menu"/>
    <addaction name="actionRotate"/>
    <addaction name="actionScaleUp"/>
    <addaction name="actionScaleDown"/>
   </widget>
   <widget class="QMenu" name="menuNonGeometryOps">
    <property name="title">
     <string>非几何变换</string>
    </property>
    <addaction name="actionScaleGrayExp"/>
    <addaction name="separator"/>
    <addaction name="actionHistogramEqualize"/>
   </widget>
   <widget class="QMenu" name="menuFFTOps">
    <property name="title">
     <string>FFT变换</string>
    </property>
    <addaction name="actionFFTOps"/>
    <addaction name="actionFFTAndIFFTOps"/>
   </widget>
   <widget class="QMenu" name="menuFilterOps">
    <property name="title">
     <string>滤波</string>
    </property>
    <addaction name="actionMeanFilter"/>
    <addaction name="actionMedianFilter"/>
    <addaction name="actionGaussianFilter"/>
   </widget>
   <widget class="QMenu" name="menuAdvanceOps">
    <property name="title">
     <string>高级应用</string>
    </property>
    <widget class="QMenu" name="menuEdgeDetect">
     <property name="title">
      <string>边缘检测</string>
     </property>
     <addaction name="actionRobertsDetect"/>
     <addaction name="actionSobelDetect"/>
     <addaction name="actionPrewittDetect"/>
     <addaction name="actionLaplaceDetect"/>
    </widget>
    <widget class="QMenu" name="menu_2">
     <property name="title">
      <string>项数重构例子</string>
     </property>
     <addaction name="actionDisplayRExample"/>
     <addaction name="actionExampleRefactor"/>
    </widget>
    <addaction name="menuEdgeDetect"/>
    <addaction name="actionKmeansClustering"/>
    <addaction name="menu_2"/>
    <addaction name="actionLaplaceSharpen"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuArithm"/>
   <addaction name="menuGeometryOps"/>
   <addaction name="menuNonGeometryOps"/>
   <addaction name="menuFFTOps"/>
   <addaction name="menuFilterOps"/>
   <addaction name="menuAdvanceOps"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionOpenFile">
   <property name="text">
    <string>打开文件</string>
   </property>
  </action>
  <action name="actionCloseFile">
   <property name="text">
    <string>关闭文件</string>
   </property>
  </action>
  <action name="actionMeanFilter">
   <property name="text">
    <string>均值滤波</string>
   </property>
  </action>
  <action name="actionSaveFile">
   <property name="text">
    <string>保存文件</string>
   </property>
  </action>
  <action name="actionMedianFilter">
   <property name="text">
    <string>中值滤波</string>
   </property>
  </action>
  <action name="actionRobertsDetect">
   <property name="text">
    <string>Roberts算子</string>
   </property>
  </action>
  <action name="actionSobelDetect">
   <property name="text">
    <string>Sobel算子</string>
   </property>
  </action>
  <action name="actionPrewittDetect">
   <property name="text">
    <string>Prewitt算子</string>
   </property>
  </action>
  <action name="actionLaplaceDetect">
   <property name="text">
    <string>Laplace算子</string>
   </property>
  </action>
  <action name="actionGaussianFilter">
   <property name="text">
    <string>高斯滤波</string>
   </property>
  </action>
  <action name="actionNegOps">
   <property name="text">
    <string>求反</string>
   </property>
  </action>
  <action name="actionToGrayScale">
   <property name="text">
    <string>灰度图像转换</string>
   </property>
  </action>
  <action name="actionToBinary">
   <property name="text">
    <string>二值图像转换</string>
   </property>
  </action>
  <action name="actionScaleGrayExp">
   <property name="text">
    <string>灰度指数级变换</string>
   </property>
  </action>
  <action name="actionHistgramEqualize">
   <property name="text">
    <string>HistgramEqualize</string>
   </property>
  </action>
  <action name="actionHistogramEqualize">
   <property name="text">
    <string>直方图均衡化</string>
   </property>
  </action>
  <action name="actionFFTOps">
   <property name="text">
    <string>快速傅里叶变换(FFT)</string>
   </property>
  </action>
  <action name="actionFFTAndIFFTOps">
   <property name="text">
    <string>FFT然后IFFT</string>
   </property>
  </action>
  <action name="actionArithmOps">
   <property name="text">
    <string>算数运算</string>
   </property>
  </action>
  <action name="actionArithmOpses">
   <property name="text">
    <string>算数运算</string>
   </property>
  </action>
  <action name="actionMove">
   <property name="text">
    <string>平移</string>
   </property>
  </action>
  <action name="actionRotate">
   <property name="text">
    <string>旋转</string>
   </property>
  </action>
  <action name="actionScaleUp">
   <property name="text">
    <string>放大</string>
   </property>
  </action>
  <action name="actionScaleDown">
   <property name="text">
    <string>缩小</string>
   </property>
  </action>
  <action name="actionFlipHorizontal">
   <property name="text">
    <string>水平镜像</string>
   </property>
  </action>
  <action name="actionFlipVertical">
   <property name="text">
    <string>垂直镜像</string>
   </property>
  </action>
  <action name="actionKmeansClustering">
   <property name="text">
    <string>K均值像素聚类</string>
   </property>
  </action>
  <action name="actionDisplayRExample">
   <property name="text">
    <string>
显示原目标边界</string>
   </property>
  </action>
  <action name="actionExampleRefactor">
   <property name="text">
    <string>边界傅里叶描述子项数重构</string>
   </property>
  </action>
  <action name="actionLaplaceSharpen">
   <property name="text">
    <string>Laplace 锐化</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="sources.qrc"/>
 </resources>
 <connections/>
</ui>
